sycl/source/detail/sampler_impl.cpp

```diff
@@ -36,18 +36,33 @@ sampler_impl::sampler_impl(cl_sampler clSampler, const context &syclContext)
 
 sampler_impl::~sampler_impl() {
   if (m_ReleaseSampler)
-    CHECK_OCL_CODE(clReleaseSampler(m_Sampler));
+    CHECK_OCL_CODE_NO_EXC(clReleaseSampler(m_Sampler));
+  // TODO replace CHECK_OCL_CODE_NO_EXC to CHECK_OCL_CODE and
+  // TODO catch an exception and add it to the list of asynchronous exceptions
 }
 
 cl_sampler sampler_impl::getOrCreateSampler(const context &Context) {
   cl_int errcode_ret = CL_SUCCESS;
   if (m_contextToSampler[Context])
     return m_contextToSampler[Context];
 
+#ifdef CL_VERSION_2_0
```

Comment:
what if CL_VERSION_2_1 or CL_VERSION_2_2 are defined? It should be also go to this branch but it does not.
Can you use CL_TARGET_OPENCL_VERSION > 120 instead here?

Reply:
Agreed. 
Done.